api/src/main/java/org/hawkular/alerts/api/services/LightweightEngine.java

```diff
@@ -6,31 +6,59 @@
 import org.hawkular.alerts.api.model.trigger.Trigger;
 
 import java.util.Collection;
+import java.util.Set;
+import java.util.UUID;
 
 /**
- * Simple engine implementation that does not rely on Infinispan or Drools for the Kafka messages processing. This is
- * the first step of a more important work aimed at making policies-engine simpler.
+ * Simple engine implementation that does not rely on Infinispan or Drools for the Kafka messages processing.
  */
 public interface LightweightEngine {
 
+    /**
+     * Loads all policies from the DB and converts them into full triggers that are stored in memory by the engine.
```

Comment:
I'm wondering, would it be possible as a next step to load the triggers on demand (i.e. when they are needed). 
I guess that would break the remaining link between the policies-engine and the policies-ui-backend.

Reply:
We could load all the triggers of a tenant/account on demand. But we can't load triggers on demand individually because when we receive a message from a tenant, we have to test all his triggers.

Reply:
Yes, I was wondering the former. Loading all the triggers of a tenant on demand (to test them all).